mlx/ops.cpp

```diff
@@ -2698,6 +2698,39 @@ array gather(
       inputs);
 }
 
+array kron(array& a, array& b, StreamOrDevice s /* = {} */) {
+  auto out_type = promote_types(a.dtype(), b.dtype());
+
+  if (a.size() == 0 || b.size() == 0) {
+    throw std::invalid_argument(
+        "Input arrays cannot be empty."); // Or your mx equivalent
+  }
+  while (a.ndim() < b.ndim()) {
+    a = expand_dims(a, {0}, s);
+  }
+  while (b.ndim() < a.ndim()) {
+    b = expand_dims(b, {0}, s);
+  }
```

Comment:
The above while loops become obsolete if you initialize the shapes with 1s as below.